pkg/asset/agent/image/ignition.go

```diff
@@ -411,10 +411,12 @@ func getRendezvousHostEnv(serviceProtocol, nodeZeroIP string, workflowType workf
 	return fmt.Sprintf(`NODE_ZERO_IP=%s
 SERVICE_BASE_URL=%s
 IMAGE_SERVICE_BASE_URL=%s
+PULL_SECRET_TOKEN=%s
 WORKFLOW_TYPE=%s
-`, nodeZeroIP, serviceBaseURL.String(), imageServiceBaseURL.String(), workflowType)
+`, nodeZeroIP, serviceBaseURL.String(), imageServiceBaseURL.String(), token, workflowType)
 }
 
+// Probably also need to add PULL_SECRET_TOKEN for authentication.
```

Comment:
@rwsu WDYT? I think the token will be needed for the nodes making API requests to assisted service once the new auth type is enabled.

Reply:
I don't think it's required here because [both](https://github.com/openshift/installer/blob/master/data/data/agent/systemd/units/agent-register-cluster.service.template#L13) [services](https://github.com/openshift/installer/blob/master/data/data/agent/systemd/units/agent-add-node.service#L14) already include `rendezvous-host.env` where you already added it above.